File: frontend/TideGuard/TideGuard/Model/WeatherData.swift
//
//  WeatherData.swift
//  TideGuard
//
//  Created by Faki Doosuur Doris on 08.05.2025.
//

import Foundation

struct WeatherData: Codable {
    let description: String?
    let temperature: Double?
    let humidity: Double?
    let imageUrl: String?
    let precipitation: Double?
    let weeklyForecast: [DailyForecast]?

    struct DailyForecast: Codable {
        let date: String?
        let maxTemp: Double?
        let minTemp: Double?
        let weatherCode: Int?
        let precipitation: Double?
        let description: String?
        let icon: String?
    }

}

File: backend/TideGuard/src/main/java/com/example/tideguard/Models/WeatherData.java
package com.example.tideguard.Models;

import lombok.Data;

import java.util.List;

@Data
public class WeatherData {

    private String description;
    private Double temperature;
    private Double humidity;
    private String imageUrl;
    private Double precipitation;

    private List<DailyForecast> weeklyForecast;

    @Data
    public static class DailyForecast {
        private String date;
        private double maxTemp;
        private double minTemp;
        private int weatherCode;
        private double precipitation;
        private String description;
        private String icon;
    }
}
